
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [23]:
import pandas as pd
import numpy as np
from scipy import stats as st

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [6]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

# What test is appropriate for this problem? Does CLT apply?


In [8]:
sum(data[data.race=='w'].call)

235.0

In [9]:
len(data[data.race == 'w'])

2435

In [10]:
len(data[data.race == 'b'])

2435

In [11]:
len(data)

4870

### We are dealing with proportions here - whether the proportion of white callbacks exceeds the proportion of black callbacks or not. Since the number of successes and failures for both races exceeds 10 and samples are independent, CLT applies and a z-test may be used.

# What are the null and alternate hypotheses?
### H<sub>0</sub>: The proportion of white callbacks is equivalent to the proportion of black callbacks.
### H<sub>A</sub>: The proportion of white callbacks is greater than the proportion of black callbacks.

# Compute margin of error, confidence interval, and p-value.

In [13]:
w = data[data.race == 'w']
b = data[data.race == 'b']

In [14]:
wp = sum(w.call)/len(w)
wp

0.096509240246406572

In [16]:
bp = sum(b.call)/len(b)
bp

0.064476386036960986

In [17]:
tp = sum(data.call)/len(data)
tp

0.080492813141683772

In [18]:
dp = wp - bp
dp

0.032032854209445585

In [20]:
se = (tp * (1- tp) * (1/len(w) + 1/len(b))) ** 0.5
se

0.0077968940361704568

In [22]:
z = dp/se
z

4.1084121524343464

In [40]:
lci = dp - st.norm.ppf(.975) * se
lci

0.016751222707276352

In [41]:
uci = dp + st.norm.ppf(.975) * se
uci

0.047314485711614819

In [42]:
# margin of error
me = (uci - lci)/2
me

0.015281631502169234

In [43]:
# confidence interval
ci = str(dp) + " +/- " + str(me) + " or [" + str(lci) + ", " + str(uci) + "]"
ci

'0.0320328542094 +/- 0.0152816315022 or [0.0167512227073, 0.0473144857116]'

In [44]:
# p-value
p = 1 - st.norm.cdf(z)
p

1.9919434187887219e-05

# Discuss statistical significance.

### The p-value of 1.99 * 10<sup>-5</sup> is less than the alpha level of 0.05, meaning there is a statistically significant increase in proportion of callbacks received by whites than blacks.